src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -29,6 +33,8 @@ const useFetchChannelHook = () => {
       topics: isAnonymous ? 'ANON_TOPIC' : 'TOPIC'
     };
 
+    const newChannel = {...channel};
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The creation of `newChannel` as a copy of `channel` seems unnecessary here. You can directly modify the properties of `channel` without affecting the original object since it's passed by value, not reference in JavaScript.



<!-- This is an auto-generated comment by OSS CodeRabbit -->